src/lib/useTableResize.ts

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import type {Column, HandleResize} from './types';
+
+export type TableColumnsWidthSetup = Record<string, number>;
```

Comment:
May be it is better to name it `ColumnWidthByName`/`WidthByColname` /`WidthByName` (not an issue).